pkg/cloudprovider/aws/node.go

```diff
@@ -34,38 +34,27 @@ type NodeFactory struct {
 }
 
 // For a given set of instanceIDs return a map of instanceID to Kubernetes node object.
-func (n *NodeFactory) For(ctx context.Context, instanceIDs []*string) (map[string]*v1.Node, error) {
-	// EC2 will return all instances if unspecified, so we must short circuit
-	if len(instanceIDs) == 0 {
-		return nil, nil
-	}
-	describeInstancesOutput, err := n.ec2api.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{InstanceIds: instanceIDs})
-	if err == nil {
-		return n.nodesFrom(describeInstancesOutput.Reservations), nil
-	}
+func (n *NodeFactory) For(ctx context.Context, instanceId *string) (*v1.Node, error) {
+	describeInstancesOutput, err := n.ec2api.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{InstanceIds: []*string{instanceId}})
```

Comment:
What happens if the instanceID is nil, does this command fail, and is that okay?

Reply:
Since it's not longer a list of instanceIDs and just a single one, our logical flow can never call this with []. The previous check was an awkward side effect of how the logic looped, without ever checking that we actually got an instance. Now that it's a single instance, we don't have this case.